File: c-programming/bitwise_operator_in_c/swap.c
#include<stdio.h>
/**
 * main - swap two integer
 * Return: 0 always (success)
 */

int main(void)
{
   int a, b, tmp;

    printf("Enter two positive integers: ");
    scanf("%d %d", &a, &b);

    printf("Before swap a = %d b = %d\n", a, b);
    tmp = a ^ b;
    a = tmp ^ a;
    b = tmp ^ b;
    printf("After  swap a = %d b = %d\n", a, b);

    return 0;
}

File: problem_solving_for_beginners/bitwise/python/swap.py
#!/usr/bin/python3
"""Swap two integers using bitwise XOR operator"""

num1, num2 = input("Enter two positive integers: ").split()

print("............Before swapping..........")
print(num1, num2)

num1 = int(num1)
num2 = int(num2)

num1 = num1 ^ num2
num2 = num1 ^ num2
num1 = num1 ^ num2


print("............After swapping..........")
print(num1, num2)